### ISSUE TITLE
Unix domain sockets

### ISSUE THREAD

**Issue by TyrfingMjolnir** [vapor/vapor#2226]
Unix domain sockets are not yet implemented in Vapor.

I would like to use unix domain sockets in order to decreasing the network prerequisits of my project.

### Steps to reproduce

```
enum Bind {
    case hostname(_: String, port: Int)
    case unixDomainSocketPath(_:String)
}
```

### Expected behavior
/opt/mymicroservice --unixdomainsocket /tmp/mymicroservice

My micro service written in Vapor will become available through unix domain socket opposed to through TCP/IP stack.
### Actual behavior

It's a waste of resources to load the network stack for TCP/IP when unix domain socket is a valid option.

### Environment

Darwin devMB15R.local 19.3.0 Darwin Kernel Version 19.3.0: Thu Jan  9 20:58:23 PST 2020; root:xnu-6153.81.5~1/RELEASE_X86_64 x86_64

-bash: vapor: command not found

* Vapor Framework version: 4.0.rc3.1
* Vapor Toolbox version: 5.2
* OS version: MacOS 10.15

Prior art: https://github.com/TyrfingMjolnir/swift-talk-backend/blob/master/Sources/NIOWrapper/NIOWrapper.swift#L266

**Comment by dimitribouniol** [comment]
I have a working implementation of this in my local branch: https://github.com/dimitribouniol/vapor/tree/unix-domain-sockets

@tanner0101 Should I create a PR for this?

**Comment by MrLotU** [comment]
Hiya @dimitribouniol, yes, please create a PR! Any comments/requests/improvements can be discussed there than 😄

**Comment by Joannis** [comment]
This is available since Vapor [4.29.0](https://github.com/vapor/vapor/releases/tag/4.29.0)
